internal/storage/storage.go

```diff
@@ -3,59 +3,172 @@ package storage
 import (
 	"awesomeProject/internal/config"
 	"context"
+	"errors"
 	"fmt"
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/mongodb" // Импорт драйвера MongoDB
+	_ "github.com/golang-migrate/migrate/v4/source/file"      // Импорт драйвера файловой системы
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 	"log/slog"
 )
 
 type Storage struct {
-	db *mongo.Collection
+	db  *mongo.Collection
+	ctx context.Context
 }
 
-type StorageFile struct {
-	ID    int
+type File struct {
 	Alias string
 	URL   string
 }
 
-func ConnectingToDB(CollectionName string, cfg *config.Config, logger *slog.Logger) (*Storage, error) {
-	const op = "storage.mongoDB.ConnectingToDB"
+var (
+	ErrURLExists   = errors.New("URL already exists")
+	ErrURLNotFound = errors.New("URL not found")
+	ZeroID         primitive.ObjectID
+)
+
+func ConnectToDB(CollectionName string, DBName string, cfg *config.Config, logger *slog.Logger, ctx context.Context) (*Storage, error) {
```

Comment:
глобальные параметры ни к чему.
```suggestion
func ConnectToDB(сollectionName string, dbName string, cfg *config.Config, logger *slog.Logger, ctx context.Context) (*Storage, error) {
```